File: src/app/shared/service/api/entryProduct.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpRequest } from '@angular/common/http';
import { EntryProductDTORequest } from '../../model/request/entryProductDTORequest';
import { environment } from 'src/environments/environment';

@Injectable({
    providedIn:'root'
})
export class EntryProductService{

    constructor(
        protected _httpClient: HttpClient
    ){}

    save(dto: EntryProductDTORequest) {
        const requ = new HttpRequest('POST', `${environment.urlBase}/entries-products/entry-product`, dto, {
            reportProgress: true,
            responseType: 'json'
        });
        return this._httpClient.request(requ);
    }
}

File: src/main/java/com/sys/botica/crce/pe/sys_botica/service/EntryProductService.java
package com.sys.botica.crce.pe.sys_botica.service;

import com.sys.botica.crce.pe.sys_botica.dto.HrefEntityDTO;
import com.sys.botica.crce.pe.sys_botica.dto.request.EntryProductDTORequest;

public interface EntryProductService {
	public HrefEntityDTO save(EntryProductDTORequest dto,String username);
}
